lib/pbench/agent/tool_meister.py

```diff
@@ -1978,41 +1956,75 @@ def main(argv):
             return 4
 
     try:
-        redis_server = redis.Redis(host=redis_host, port=redis_port, db=0)
+        redis_server = redis.Redis(host=parsed.host, port=parsed.port, db=0)
     except Exception as exc:
         print(
-            f"{PROG}: Unable to construct Redis client, {redis_host}:{redis_port}: {exc}",
+            f"{PROG}: Unable to construct Redis client, {parsed.host}:{parsed.port}: {exc}",
             file=sys.stderr,
         )
         return 5
 
     try:
         # Wait for the key to show up with a value.
-        params_str = wait_for_conn_and_key(redis_server, param_key, PROG)
+        params_str = wait_for_conn_and_key(redis_server, parsed.key, PROG)
         params = json.loads(params_str)
         # Validate the tool meister parameters without constructing an object
         # just yet, as we want to make sure we can talk to the redis server
         # before we go through the trouble of daemonizing below.
         ToolMeister.fetch_params(params)
     except Exception as exc:
         print(
-            f"{PROG}: Unable to fetch and decode parameter key, '{param_key}': {exc}",
+            f"{PROG}: Unable to fetch and decode parameter key, '{parsed.key}': {exc}",
             file=sys.stderr,
         )
         return 6
 
-    func_args = (
+    func = daemon if parsed.daemonize else driver
+    func(
         PROG,
         tar_path,
         sysinfo_dump,
         pbench_install_dir,
         tmp_dir,
-        param_key,
+        parsed,
         params,
         redis_server,
     )
-    if daemonize == "yes":
-        ret_val = daemon(*func_args, redis_host, redis_port)
+
+
+def main(argv):
+    """Main program for the Tool Meister.
+
+    Arguments:  argv - a list of parameters
+
+                argv[1] - host name or IP address of Redis Server
+                argv[2] - port number of Redis Server
+                argv[3] - name of key in Redis Server for operational
+                          parameters
+                argv[4] - "yes" to run as a daemon
+                argv[5] - desired debug level
+
+    Returns 0 on success, > 0 when an error occurs.
+    """
+    prog = Path(argv[0])
+    PROG = prog.name
+    try:
+        parsed = Arguments(
+            host=argv[1],
+            port=int(argv[2]),
+            key=argv[3],
+            daemonize=argv[4] == "yes" if len(argv) > 4 else False,
+            level=argv[5] if len(argv) > 5 else "info",
+        )
+    except (ValueError, IndexError) as e:
+        print(f"{PROG}: Invalid arguments, {argv!r}: {e}", file=sys.stderr)
+        return 1
```

Comment:
As I mentioned above, we should have a catch-all clause which logs the traceback and prints only the exception message, rather than dying and dumping the traceback to the terminal.

Reply:
These aren't CLI commands; they're only run via `ssh` from `pbench-tool-meister-start` and any output (except through the `logger` once we hook up the `RedisHandler`) is completely lost anyway. I'd like to work on that, and originally intended to at least start addressing it in this PR, but as mentioned I scaled it back to minimize conflicts. (And, yes, I didn't undo my main/start split so I wasn't entirely consistent in following through on that decision, but under the circumstances I don't feel inclined to add another level of `try`/`except` for this comment, in this context.)

Reply:
Ha!  Since the traceback will otherwise be lost, as you say, it's important then to catch the exception and properly log it....  😆

Seriously, though, I think that all I'm suggesting here is one additional `except` clause which catches all the other exceptions and logs them.  But, if there's another PR coming for this stuff, I can retract this comment....